runtimes/dart-2.18/server.dart

```diff
@@ -10,56 +10,121 @@ import 'function_types.dart';
 
 void main() async {
   await shelf_io.serve((req) async {
-    List<String> userLogs = [];
-    if (req.method != 'POST') {
-      return shelf.Response(500, body: 'Invalid request');
+    if ((req.headers['x-open-runtimes-secret'] ?? '') == '' || (req.headers['x-open-runtimes-secret'] ?? '') != (Platform.environment['OPEN_RUNTIMES_SECRET'] ?? '')) {
+      return shelf.Response(500, body: 'Unauthorized. Provide correct "x-open-runtimes-secret" header.');
     }
-    if (req.headers['x-internal-challenge'] !=
-        Platform.environment['INTERNAL_RUNTIME_KEY']) {
-      return shelf.Response(500, body: 'Unauthorized');
+
+    String bodyString = await req.readAsString();
+    dynamic body = bodyString;
+    String method = req.method;
+    Map<String, dynamic> headers = {};
+
+    for (MapEntry entry in req.headers.entries) {
+      String header = entry.key.toLowerCase();
+      if(!header.startsWith('x-open-runtimes-')) {
+        headers[header] = entry.value;
+      }
+    }
+
+    String contentType = req.headers['content-type'] ?? 'plain/text';
+    if(contentType.contains('application/json')) {
+      if(!bodyString.isEmpty) {
+        body = jsonDecode(bodyString);
+      } else {
+        body = {};
+      }
+    }
+
+    String hostHeader = req.headers['host'] ?? '';
+
+    String path = '/' + req.url.path;
```

Comment:
I think we should be using `req.requestedUri` instead. debugger shows the expected values in `req.requestedUri`:

<img width="264" alt="image" src="https://user-images.githubusercontent.com/1477010/221075845-9ba71fea-b235-4ceb-a7ef-1c3ef7ef22d7.png">